File: src/main/java/adventOfCode/lib/Solver.java
package adventOfCode.lib;

public interface Solver {
    public String getYearStr();

    public String getDayStr();

    public default String solve(String input) {
        Number part1Ans = this.part1(input);
        Number part2Ans = this.part2(input);

        return this.getYearStr() + " Day " + this.getDayStr() +
                ":\n  Part 1: " + part1Ans + "\n  Part 2: " + part2Ans;
    }

    public Number part1(String input);

    public Number part2(String input);
}

File: src/main/kotlin/adventOfCode/lib/Solver.kt
package adventOfCode.lib

interface Solver {
    val year: String
    val day: String

    fun solve(input: String) {
        val part1Ans = part1(input)
        val part2Ans = part2(input)

        println("${this.day} Day ${this.year}:\n  Part 1: $part1Ans\n  Part 2: $part2Ans")
    }

    fun part1(input: String): Number

    fun part2(input: String): Number
}
